### PR TITLE
Attempt to hover/select without element

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Removes `div style="display:contents"` arount Radix etc Components.

Below in Implementation is written why SelectedInstanceConnector was moved into standalone root level hook.
In short
- [x] - No issues with visually hidden components.
- [x] - Allows us to not use refs for the builder support. (Now our components can be without refs)
- [ ] - cc @TrySound ContentEditable could also use getAllElementsBoundingBox from domUtils to get element ref this will allow to remove useRef from WebstudioDevComponent

Changes from previous behavior:

- Now on preview elements are not selected (now is like in webflow).
- On changing marign etc we do not hide outline as before

## Implementation: 

having `instanceSelector` we can 
- Find all elements with attribute `data-ws-selector` which ends with `instanceSelector`
- Then find "root/parent" elements of found elements. _(We can find few subtrees)_
- Then use that elements instead of wrapper element.

This works fine with tree hovering and selections.
This allows us to not use `refs` inside our WebstudioDevComponent at all.

### Below is why SelectedInstanceConnector was rewritten

The issue arises when such element is duplicated by radix into VisuallyHidden element. 

There is no way to detect now inside `WebstudioComponentDev` does current Component is inside  VisuallyHidden
(We have no reference element to detect that.)

So current approach makes any additional changes useless.
I'm going to close this. And leave just for example

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/canvas/canvas.tsx [hunk 1]
```diff
@@ -52,6 +52,7 @@ import { subscribeInstanceSelection } from "./instance-selection";
 import { subscribeInstanceHovering } from "./instance-hovering";
 import { useHashLinkSync } from "~/shared/pages";
 import { useMount } from "~/shared/hook-utils/use-mount";
+import { useSelectedInstance } from "./features/webstudio-component/selected-instance-connector";
```
Comment (kof) [inline apps/builder/app/canvas/canvas.tsx:55]: shouldn't webstudio-component have an index and import from the index?

Comment (kof) [reply apps/builder/app/canvas/canvas.tsx:55]: it already has an index

File: apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx [hunk 2]
```diff
@@ -32,12 +32,11 @@ import {
   type InstanceSelector,
   areInstanceSelectorsEqual,
 } from "~/shared/tree-utils";
-import { SelectedInstanceConnector } from "./selected-instance-connector";
+// import { SelectedInstanceConnector } from "./selected-instance-connector";
```
Comment (TrySound) [inline apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx:35]: remove comment?

File: apps/builder/app/canvas/instance-hovering.ts [hunk 3]
```diff
@@ -44,27 +45,41 @@ export const subscribeInstanceHovering = () => {
       hoveredInstanceSelectorStore.set(undefined);
       hoveredInstanceOutlineStore.set(undefined);
     }, 100);
+
+    // Fixes the bug, that new hover occures during timeout
```
Comment (TrySound) [inline apps/builder/app/canvas/instance-hovering.ts:49]: Not a bug if it is fixed

Comment (istarkov) [reply apps/builder/app/canvas/instance-hovering.ts:49]: ? The code Fixes the bug
What the issue?

File: apps/builder/app/canvas/instance-hovering.ts [hunk 4]
```diff
@@ -44,27 +45,41 @@ export const subscribeInstanceHovering = () => {
...
-    if (instanceId === undefined) {
+  const updateHoveredRect = (
+    elements: Element[],
+    instanceSelector: Readonly<InstanceSelector>
+  ) => {
+    if (elements.length === 0) {
       return;
     }
+
+    if (instanceSelector.length === 0) {
```
Comment (TrySound) [inline apps/builder/app/canvas/instance-hovering.ts:68]: The utility protects from such case by returning undefined. No need to check here.

File: apps/builder/app/canvas/instance-hovering.ts [hunk 5]
```diff
@@ -44,27 +45,41 @@ export const subscribeInstanceHovering = () => {
...
+      return;
+    }
+
+    const instanceId = instanceSelector[0];
```
Comment (TrySound) [inline apps/builder/app/canvas/instance-hovering.ts:72]: const [instanceId] = instanceSelector

File: apps/builder/app/canvas/instance-hovering.ts (hunk 4 above)
Comment (istarkov) [reply apps/builder/app/canvas/instance-hovering.ts:68]: Until I see type 
type NonEmptyArray<T> = [T, ...T[]]
I would check length, as its some internal implementation detail of selector not exposed somehow

**Review by TrySound — APPROVED** [review]

**Pull request merged**